### DISCUSSION TITLE
Command Palette Additions

### DISCUSSION THREAD

**Discussion by Maximinodotpy** [godotengine/godot-proposals#6471] [Editor]
I myself think the new Command Palette is an awesome addition as I really like to use my Keyboard for most tasks, but as soon as I started using it I noticed that some really important commands are missing and that I still had to use the mouse. So now I want start a discussion for more commands that should be added the command palette to make it the best it can be 😀!

Below you see a list of commands that I think are missing or could be added.

- Focus to ...
  - Import Panel
  - Scene Panel
  - FileSystem Panel
  - Inspector Panel
  - Node Panel
  - History Panel
- Create Node / Instantiate Scene [I know there is `ctrl + a` and `ctrl + shift + a` but I think they should be added for completeness]
- *[ Focused On Any Panel ]*
  - Break out current panel as seperate Window
- *[ Focused on Node or FileSystem Panel ]*
  - Open Context Menu on Node / File
- Export as ...
- Open Plugins Manager
- Open Input Mappings
- New Resource
- Open Userdata Path in File Explorer
- Open Project in File Explorerer
- Reload Current Project
- Focus Console
- Clear Console
- Open Debugger
  - And any submenu

I am not sure but plugins and tool scripts should be able to add commands.

these are just some suggestions, what do you have in mind?

**Comment by thekeith** [comment]
Been a few years since someone chimed in here - I'm going to jump in and see if I can register the whole command API actually. I'd very much like to have a full exposure of the command API in the palette (though possibly behind an option like "advanced command palette")

**Comment by kitbdev** [comment]
There are commands for focusing on the various docks, like `Open Import Dock` or `Toggle Output Dock` and there is a PR for reload project https://github.com/godotengine/godot/pull/106004
